### PR TITLE
translate Refs and the DOM

### PR DESCRIPTION
**Pull request by SafaElmali**

### CODE REVIEW

**Comment by alioguzhan** [comment]
Hey, thanks for the contribution. 

Can you please read the basic rules on #1 for contribution and commit message styles. Also, it would be better if you leave a comment below with the name of the page you would like to translate so that we can add you to the list.

Thanks.

**Comment by alioguzhan** [comment]
Hey @SafaElmali , any update on this? If you don't have time we can re-assign it to someone else. Thanks

**Comment by SafaElmali** [comment]
> Hey @SafaElmali , any update on this? If you don't have time we can re-assign it to someone else. Thanks

I will commit it soon. Sorry for delay.

**Comment by SafaElmali** [comment]
Could you please check the missing parts? I wasn't sure for those parts. That's why I didn't translate. @alioguzhan

**Comment by alioguzhan** [comment]
> Could you please check the missing parts? I wasn't sure for those parts. That's why I didn't translate. @alioguzhan

Ok, I will check those parts in a few days. I'll let you know. Thanks

File: content/docs/refs-and-the-dom.md [hunk 1]
```diff
@@ -275,16 +285,18 @@ class Parent extends React.Component {
...
 
-> Note
->
-> If you're currently using `this.refs.textInput` to access refs, we recommend using either the [callback pattern](#callback-refs) or the [`createRef` API](#creating-refs) instead.
+
+> Not
+> 
+> Eğer `this.refs.textInput`'u reflere erişmek için kullanıyorsanız, [`createRef` API](#creating-refs) veya [callback refleri](#callback-refs)nden birini kullanmanızı tavsiye ederiz.
 
 ### Caveats with callback refs {#caveats-with-callback-refs}
```
Comment (alioguzhan) [inline content/docs/refs-and-the-dom.md]: > Callback Ref'lerine Dair Uyarılar

File: content/docs/refs-and-the-dom.md [hunk 2]
```diff
@@ -275,16 +285,18 @@ class Parent extends React.Component {
...
 If you worked with React before, you might be familiar with an older API where the `ref` attribute is a string, like `"textInput"`, and the DOM node is accessed as `this.refs.textInput`. We advise against it because string refs have [some issues](https://github.com/facebook/react/pull/8333#issuecomment-271648615), are considered legacy, and **are likely to be removed in one of the future releases**. 
+Daha önceden React ile uğraştıysanız, `ref` özelliğinin `"textInput"` gibi bir string olduğu eski API'ya aşina olabilirsiniz ve DOM node'una da `this.refs.textInput` şeklinde erişilirdi. [Bazı sorunlar](https://github.com/facebook/react/pull/8333#issuecomment-271648615) are considered legacy ve **gelecek sürümlerden birinde kaldırılması muhtemeldir**. 
 
-> Note
->
-> If you're currently using `this.refs.textInput` to access refs, we recommend using either the [callback pattern](#callback-refs) or the [`createRef` API](#creating-refs) instead.
+
+> Not
+> 
+> Eğer `this.refs.textInput`'u reflere erişmek için kullanıyorsanız, [`createRef` API](#creating-refs) veya [callback refleri](#callback-refs)nden birini kullanmanızı tavsiye ederiz.
```
Comment (alioguzhan) [inline content/docs/refs-and-the-dom.md]: Semantic fix: add seçeneklerinden
> Eğer `this.refs.textInput`'u reflere erişmek için kullanıyorsanız, [`createRef` API](#creating-refs) veya [callback refleri](#callback-refs) seçeneklerinden birini kullanmanızı tavsiye ederiz.

File: content/docs/refs-and-the-dom.md [hunk 3]
```diff
@@ -275,16 +285,18 @@ class Parent extends React.Component {
...
 
 ### Caveats with callback refs {#caveats-with-callback-refs}
 
-If the `ref` callback is defined as an inline function, it will get called twice during updates, first with `null` and then again with the DOM element. This is because a new instance of the function is created with each render, so React needs to clear the old ref and set up the new one. You can avoid this by defining the `ref` callback as a bound method on the class, but note that it shouldn't matter in most cases.
+Eğer `ref` callback bir inline fonksiyon olarak tanımlanmışsa, güncelleme anında iki kez çağırılacaktır. İlk olarak `null` ve daha sonrasında DOM elemanı yeniden çağrılır.. Bunun sebebi, fonksiyonun bir nesnesi her render'da oluşturulur. Bu yüzden React eski ref'i kaldırır ve yenisini ekler. Bunu önlemek için  `ref` callback'ini sınıfa bağlı bir method olarak tanımlayabilirsiniz. Ancak bu birçok durumda önemli değildir.
```
Comment (alioguzhan) [inline content/docs/refs-and-the-dom.md]: > inline fonksiyon -> satıriçi fonksiyon

Better:
> İlk olarak `null` ile, sonrasında DOM elemanı ile yeniden çağrılır.

Turkish version:
> method -> metod

Comma after `bu`:
> `Ancak bu bircok` -> `Ancak bu, bircok`

File: content/docs/refs-and-the-dom.md [hunk 4]
```diff
@@ -275,16 +285,18 @@ class Parent extends React.Component {
...
+Yukarıda ki örnekte, `Parent`, ref callback'ini `inputRef` prop'u olarak `CustomTextInput`una aktarır ve `CustomTextInput`u aynı fonksiyonu özel bir `ref` özelliği olarak `<input>`a aktarır. Sonuç olarak, `Parent`taki `this.inputElement`i, `CustomTextInput`taki `<input>` elemanına karşılık gelen DOM node'una set edilir.
 
-### Legacy API: String Refs {#legacy-api-string-refs}
+### Eski API: String Refler {#legacy-api-string-refs}
 
 If you worked with React before, you might be familiar with an older API where the `ref` attribute is a string, like `"textInput"`, and the DOM node is accessed as `this.refs.textInput`. We advise against it because string refs have [some issues](https://github.com/facebook/react/pull/8333#issuecomment-271648615), are considered legacy, and **are likely to be removed in one of the future releases**. 
+Daha önceden React ile uğraştıysanız, `ref` özelliğinin `"textInput"` gibi bir string olduğu eski API'ya aşina olabilirsiniz ve DOM node'una da `this.refs.textInput` şeklinde erişilirdi. [Bazı sorunlar](https://github.com/facebook/react/pull/8333#issuecomment-271648615) are considered legacy ve **gelecek sürümlerden birinde kaldırılması muhtemeldir**.
```
[the remark this answers is not in the document]
Comment (SafaElmali) [reply content/docs/refs-and-the-dom.md]: Should I replace every "**DOM Nodes**" with "**DOM düğümü**" ?

[the remark this answers is not in the document]
Comment (alioguzhan) [reply content/docs/refs-and-the-dom.md]: yes you can replace each `node` with `dugum`

**Comment by alioguzhan** [comment]
You missed some parts from my last review. Can you please check all of them. Thanks! @SafaElmali

**Comment by alioguzhan** [comment]
Thanks. Merging it...

**Comment by SafaElmali** [comment]
> Thanks. Merging it...

Thank you! and sorry for my translating mistakes 😬

**Comment by alioguzhan** [comment]
> > Thanks. Merging it...
> 
> Thank you! and sorry for my translating mistakes 😬

No worries! We all are trying to contribute, with good intentions. :)

**Pull request merged**